cmd/container-manager/manage_ctr_mgr.sh

```diff
@@ -0,0 +1,330 @@
+modify_docker_service(){
+	docker_options=$1
+	docker_service_dir="/etc/systemd/system/docker.service.d/"
+	if [ "$(ls -A $docker_service_dir)" ] && [ ${force} == false ]; then
+		die "Found a service configuration file. Please use -f flag to overwrite the service configuration"
+	elif [ "$(ls -A $docker_service_dir)" ] && [ ${force} == true ]; then
+		rm -rf "${docker_service_dir}/*"
+	fi
+	sudo mkdir -p "$docker_service_dir"
+	cat <<EOF | sudo tee "$docker_service_dir/clear-containers.conf"
+[Service]
+Environment="$docker_http_proxy"
+Environment="$docker_https_proxy"
+ExecStart=
+ExecStart=/usr/bin/dockerd ${docker_options}
+EOF
+	echo "Restart docker service"
+	sudo systemctl daemon-reload
+	sudo systemctl restart docker
+}
+
+# This function configures docker to work by default with the
+# specified runtime.
+configure_docker(){
+	if [ -z "$runtime" ]; then
+		die "please specify a runtime with -r"
+	fi
+
+	if [ ! "$(info_docker)" ]; then
+		die "Docker is not installed. Please install it before configuring the runtime"
+	fi
+
+	if [ "$(get_docker_default_runtime)" == "$runtime" ]; then
+		message "configure_docker: $runtime is already configured as default runtime"
+	else
+		log_message "configure_docker: configuring $runtime as default docker runtime"
+		# Check if the system has set http[s] proxy
+		if [ -n "$http_proxy" ] && [ -n "$https_proxy" ] ;then
+			docker_http_proxy="HTTP_PROXY=$http_proxy"
+			docker_https_proxy="HTTPS_PROXY=$https_proxy"
+		fi
+
+		if [ "$runtime" == "cc-runtime" ]  ; then
+			# Try to find cc-runtime in $PATH, if it is not present
+			# then the default location will be /usr/local/bin/cc-runtime
+			cc_runtime_bin="$(which $runtime)" || \
+				die "$runtime cannot be found in $PATH, please make sure it is installed"
+			docker_options="-D --add-runtime $runtime=$cc_runtime_bin --default-runtime=$runtime --storage-driver=overlay2"
+			modify_docker_service "$docker_options"
+		elif [ "$runtime" == "runc" ]  ; then
+			docker_options="-D"
+			modify_docker_service "$docker_options"
+		else
+			die "configure_docker: runtime $runtime not supported"
+		fi
+	fi
+}
+
+main(){
+	# Check if the script is run without arguments
+	[ "$#" -eq 0 ] && usage && exit 1
+
+	# Parse Usage options:
+	while getopts ":h" opt; do
+		case ${opt} in
+			h )
+				usage
+				exit 0
+				;;
+			\? )
+				echo "Invalid Option: -$OPTARG" 1>&2
+				usage
+				exit 1
+				;;
+		esac
+	done
+	shift $((OPTIND -1))
+
+	ctr_manager=$1; shift
+	case "$ctr_manager" in
+	# Parse options
+		docker)
+			subcommand=$1; shift
+			parse_subcommand_options "$@"
+			;;
+		*)
+			warning "container manager $ctr_manager is not supported."
+			usage
+			exit 1
+	esac
+
+	shift "$((OPTIND - 1))"
+
+	case "$subcommand" in
+		configure )
+			configure_docker
+			;;
+
+		info )
+			info_docker
+			;;
+
+		install )
+			install_docker
+			;;
+
+		remove )
+			remove_docker
+			;;
+
+		*)
+			echo "Invalid subcommand: $subcommand"
```

Comment:
Might be worth quoting `$subcommand` to make it clearer what the problem is if it's blank/not specified.